tsdb/head_test.go

```diff
@@ -6363,6 +6363,154 @@ func TestHeadAppender_AppendCTZeroSample(t *testing.T) {
 	}
 }
 
+func TestHeadAppender_AppendHistogramCTZeroSample(t *testing.T) {
```

Comment:
Knowing that OOO native histograms will have more changes coming up I think we need a regression test to cover the case of ct falling behind the last ingested sample:
```
{ts:100, h: testHistogram, ct: 1}
{ts:101, h: testHistogram, ct: 1}
```
I'm surprised that there's no such test for floats .

Reply:
I've just added a new sample to the first test case :)